app/services/record_declarations/base.rb

```diff
@@ -0,0 +1,97 @@
+# frozen_string_literal: true
+
+require "json_schema/validate_body_against_schema"
+
+module RecordDeclarations
+  class Base
+    attr_accessor :params
+    delegate :user_profile, :actual_lead_provider, to: :not_implemented_error
+
+    class << self
+      def call(params)
+        new(params).call
+      end
+
+      def required_params
+        %i[participant_id cpd_lead_provider declaration_type declaration_date course_identifier raw_event]
+      end
+    end
+
+    def not_implemented_error
+      raise NotImplementedError, "Method must be implemented"
+    end
+
+    def call
+      validate_schema!
+      add_participant_profile_params!
+
+      declaration = create_record!
+      validate_provider!
+      { id: declaration.id }
+    end
+
+  private
+
+    def initialize(params)
+      @params = params
+      @params[:user_id] = params[:participant_id]
+    end
+
+    def validate_schema!
+      errors = JsonSchema::ValidateBodyAgainstSchema.call(schema: schema, body: params[:raw_event])
+      raise ActionController::ParameterMissing, (errors.map { |error| error.sub(/\sin schema.*$/, "") }) unless errors.empty?
+    end
+
+    def schema_validation_params
+      { version: "0.3" }
+    end
+
+    def schema
+      JSON.parse(File.read(JsonSchema::VersionEventFileName.call(schema_validation_params)))
+    end
+
+    def add_participant_profile_params!
```

Comment:
Is this named correctly?

Reply:
Good point, `validate_participant_params!` sounds like a much better name.